### ISSUE TITLE
Weak 결과

### ISSUE THREAD

**Issue by wonjae124** [wonjae124/2021_Lab_project#8]
<Test 1>

![image](https://user-images.githubusercontent.com/67944072/155510557-131fb1b1-a790-4308-847b-d32a3dcfa65e.png)

변경 사항 : BatchNorm2d를 총 4개 생성함.
결과 :   
장점 - Loss가 기존 0.9에서 정체되었는데 적은 epoch에도 loss가 0.35로 급격하게 줄어 들음. Global optimum을 빨리 찾게 만든다.
단점 - 빨간색 무늬가 생김. 학습 시간은 그대로임.

<Test 2>

![image](https://user-images.githubusercontent.com/67944072/155515589-2772169e-99f2-4225-9a1a-dce1c2aa36d0.png)

![image](https://user-images.githubusercontent.com/67944072/155515633-5ee031a8-a755-4201-8e0c-606296524e1b.png)

변경사항 batch_size를 16에서 32, batchNorm2d를 4개에서 2개로.
결과 :
장점 - 빨간 격자 무늬가 사라짐. 복원 잘 됨
단점 - val_loss의 변동이 심해짐 , PSNR값이 평균 20으로 많이 낮음. 아까 일부러 psnr 식에서 분자를 1로 지정 한게 원인인듯 하다. 
 배경으로는 이미지를 numpy로 바꿈으로써 이미지 값의 범위가 0-255로 변했기에 다시  최대값을 255로 설정하는게 맞는 것 아닌가? 

<Test3>

![image](https://user-images.githubusercontent.com/67944072/155529442-de577754-31a7-4683-b49e-4c4997327cba.png) 

![image](https://user-images.githubusercontent.com/67944072/155529524-748a5f1a-1dda-4532-ad53-1f6f77f2b78b.png) 

확인사항 :  복원 이미지의 값 범위는 0~1(float32). PSNR식이 분자에 3을 곱해야 함
변경사항 : Drop_out 추가 -> 화면 흑백 처리 됨, loss가 3배나 높음. 제외 시킴...
a. epoch20 -> 55
b. PSNR 두 버젼으로 만듦.

<Tesr4>

![image](https://user-images.githubusercontent.com/67944072/155530399-851fc09c-a7ce-45fc-a633-13407c5df3f0.png)

확인사항 : PSNR식에 분모에 3을 곱하는 중대한 실수/ CNN이 6X6 크기로 만드는게 격자의 원인으로 봄

변경사항 : 마지막 return 할 때, 분자에 3을 곱하는 형태 변경/CNN 통과시 img size가 12X12에서 끝나게 만듦. 계층 줄임

<Test5>

![image](https://user-images.githubusercontent.com/67944072/155536981-d3d35715-b08d-480f-902f-6efcf3ddd251.png)

확인사항 : 이번에는 점이 찍히기 시작함 .
변경사항 :  로스가 계속 떨어지는게 이상함. 모델 초기화가 필요해서 런타임 다시 시작

<Test5>

![image](https://user-images.githubusercontent.com/67944072/155537559-c89fd666-6d55-4bf7-a229-e9017e54e4fc.png)
기존 모델

![image](https://user-images.githubusercontent.com/67944072/155546234-2828f37d-ea9a-40c0-aa79-24b5e7510bc8.png)

두 배로 쌓은 모델 -> 격자 무늬가 더 심해짐. 촘촘해짐 아주,

<Final Test6>
![image](https://user-images.githubusercontent.com/67944072/155553003-b18d6e76-4276-403f-bf35-c8bbd9d4c490.png)

변경사항 : Feature 개수에 맞게 모델 축소 함. 2만 7천개보다 좀 더 많게 함. 8 by 8 image로 나오기 때문에 최종 채널을 8로 하였음. 근데 생각해보니 이미지 사이즈가 8이 되는거가 중요하겠다...

확인사항 : 모델 과적합이 너무 심하게 발생 함.
변경사항 : 모델 과적합은 재쳐두고 일단 모델 파라미터 증가 시킴
--------------------------------이론 보충-------------
1. 배치 노말
![image](https://user-images.githubusercontent.com/67944072/155513192-51fe5031-860c-4ab4-84c8-7c2c01a4b840.png)
https://hyomyo.tistory.com/59

argmin / argmax
말 그대로 arguments of min, arguments of max란 뜻입니다. 즉, 어떤 함수를 최소/최대로 만드는 정의역의 점들, elements 혹은 매개변수를 말합니다. 

예를들어서 다음 수식은 f(x)가 최소값을 갖는 x의 값을 의미합니다.

2. PNSR 구하는 식
 
![image](https://user-images.githubusercontent.com/67944072/155528141-7366f3eb-8fea-4ff0-ad21-70320ef3578b.png)
출처 : 이혜영. 컬러이미지 대상 고용량 적응형 LSB 스테가노그래피

![image](https://user-images.githubusercontent.com/67944072/155528844-8c287e5a-1ba8-4760-b8f6-e334b4db425a.png)
https://dsp.stackexchange.com/questions/71845/defining-the-snr-or-psnr-for-color-images-3-channel-rgb-files

===================

22.03.30 - 잠재변수, maxpooling을 추가 하는 것을 적용해보자!!! untill (22.04.06)
https://colab.research.google.com/drive/1Q2Hljbh4gfVWRFwwtY89fauY2Lf4jvH8#scrollTo=fY18EkZDO93a&uniqifier=2
